WcaOnRails/app/models/competition.rb

```diff
@@ -1214,6 +1218,24 @@ def update_persons_wcif!(wcif_persons, current_user)
           roles = wcif_person["roles"] - ["delegate", "organizer"] # These two are added on the fly.
           registration.update!(roles: roles)
         end
+        if registration && wcif_person["assignments"]
+          competition_activities = all_activities
+          registration.assignments = []
+          wcif_person["assignments"].each do |assignment_wcif|
+            schedule_activity = competition_activities.find do |competition_activity|
+              competition_activity.wcif_id == assignment_wcif["activityId"]
+            end
```

Comment:
So the idea here is that based on the local WCIF `activityId` we find the corresponding `ScheduleActivity` record and use its id as the reference for the Assignment. This means that we can easily request all assignments for an activity and the other way round.